# Marsden objective 1: non-destructive heDNA

## 1. Introduction

This GitHub repository serves as a comprehensive guide for the bioinformatic and statistical analysis of the ethanol comparison project, associated with the Marsden Fast-Start funding MFP-UOO002116.

## 2. Experimental design

The primary objective of this experiment is to assess the possibility of non-destructive heDNA recovery from museum-stored marine sponge specimens through DNA extraction from the ethanol in which specimens are stored. To verify non-destructive sampling as a suitable methodological choice for specimens across the phylum Porifera, we selected one specimen from the three major classes due to differing internal skeletal structures, including Hexactinellida (hexagonal silica spicules), Demospongiae (spongin), and Calcarea (calcium carbonate spicules). A detailed list with metadata for each specimen can be found below.

| Cat. # | Class | Species | Date | Latitude | Longitude | Depth |
| :---- | :---- | :---- | :---- | :---- | :---- | :---- |
| 35574 | Calcarea | Petrobiona sp. | 09/02/2008 | -73.1245 | 174.3205 | 321 |
| 37535 | Demospongia | Homaxinella sp. | 24/02/2008 | -72.0093 | 173.2238 | 850 |
| 37124 | Hexactinellida | Rossella fibulata | 21/02/2008 | -72.5903 | 175.3423 | 479 |

For each sepcimen, DNA was extracted from 10 tissue biopsies without removal of excess ethanol and 10 tissue biopsies whereby excess ethanol was removed by absorbance of lint-free kimwipes. To test heDNA recovery from ethanol, we extracted DNA through (1) 1 ml evaporation, (2) 1 ml centrifugation, (3) 1 ml precipitation, (4) 1 ml filtration, and (5) 10 ml filtration. For each non-destructive method, 10 replicates were processed to enable statistical comparison between treatments.

After DNA extraction, the total DNA concentration was measured via Qubit, while DNA purity was assessed by investigating the 260/280 and 260/230 absorbence ratios measured via Denovix. To amplify heDNA signals associated with fish species (Fish16SF: 5’-GACCCTATGGAGCTTTAGAC-3’; Fish16S2R: 5’-CGCTGTTATCCCTADRGTAACT-3’), we employed a metabarcoding approach. This primer set was chosen over the MiFish-U primer set, as it does not co-amplify human DNA, a potential contaminant signal for museum-stored specimens that have been handled repeatedly for morphometric analysis.

**While we're unsure at this stage, genome sequencing was also conducted on the ONT platform to assess the possibility of retrieving (mito) genome sequences from non-destructive methods. However, we'll have to wait until we receive the results later on if this will be included in the manuscript.**

## 3. Bioinformatic analysis

For the bioinformatic analysis of the metabarcoding data, we will employ a standard [cutadapt v4.4](https://cutadapt.readthedocs.io/en/stable/) and [VSEARCH v2.16.0_macos_aarch64](https://github.com/torognes/vsearch) pipeline, followed by taxonomy assignment through a local blastn search against a curated reference database generated by [CRABS v0.1.8](https://github.com/gjeunen/reference_database_creator), and a final data curation step using [tombRaider v1.0](https://github.com/gjeunen/tombRaider) to remove PCR artefacts.

## 3.1 Starting files

For this project, we need six files for the bioinformatic analysis, including three sequence files and three metadata files. File names are listed below. All files (sequencing and metadata) can be downloaded from [figshare](https://figshare.com/account/home#/projects/184573).

Sequence file names:

1. 8437-P1-00-01_S1_L001_R1_001.fastq.gz
2. 8482-P1-0-1_S1_L001_R1_001.fastq.gz
3. 8482-P2-00-01_S1_L001_R1_001.fastq.gz

Metadata file names:

1. 8437-P1metadata.fasta
2. 8482-P1metadata.fasta
3. 8482-P2metadata.fasta

Before we start with our analysis, let's reformat these files to make it easier for us. For example, we can merge all sequencing files and all metadata files into one file per type, as the code can then be run once, rather than in a loop. Additionally, we need to strip the `^` and the `$` from the metadata files, as this approach won't be used in our pipeline. First, let's merge the sequence files. To do this, we need to unzip the files prior to merging them.

```{code-block} bash
gunzip *.fastq.gz
cat *.fastq > ethanol_comparison_heDNA.fastq
rm -r *001.fastq
```

Next, we can reformat the metadata files using the python script below. At the same time, it will output the merged metadata file.

```{code-block} python
#!/usr/bin/env Python3
input_file_list = ['8437-P1metadata.fasta', '8482-P1metadata.fasta', '8482-P2metadata.fasta']
merged_input_files = []
for item in input_file_list:
   with open(item, 'r') as infile:
      for line in infile:
         line = line.lstrip('^').rstrip('$\n') + '\n'
         merged_input_files.append(line)
with open('ethanol_comparison_heDNA_metadata.fasta', 'w') as outfile:
   for item in merged_input_files:
      outfile.write(item)
```

## 3.2 Raw sequence data quality check

Before we get started with the bioinformatic analysis, we will check the quality and integrity of the raw sequence file using [FastQC v0.12.1](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/).

```{code-block} bash
mkdir 1.fastqc_raw
fastqc ethanol_comparison_heDNA.fastq -o 1.fastqc_raw
```

![FastQC - Raw](figures/fastqc_raw.png)

## 3.3 Demultiplexing

Once quality is checked, we can assign sequences to their respective samples, i.e., demultiplexing. For this step, we will use cutadapt. At the same time as removing the barcodes from the amplicon, we will remove primer-binding regions as well, by setting the "adapter" to the barcode sequence + primer sequence.

```{code-block} bash
mkdir 2.demultiplex
cutadapt ethanol_comparison_heDNA.fastq -g file:'ethanol_comparison_heDNA_metadata.fasta' -o 2.demultiplex/{name}.fastq --discard-untrimmed --no-indels -e 0 --cores=0
```

```{admonition}
=== Summary ===

Total reads processed:              36,349,307
Reads with adapters:                24,149,058 (66.4%)

== Read fate breakdown ==
Reads discarded as untrimmed:       12,200,249 (33.6%)
Reads written (passing filters):    24,149,058 (66.4%)

Total basepairs processed: 10,941,141,407 bp
Total written (filtered):  4,440,167,368 bp (40.6%)
```

## 3.4 Quality filtering

We can further filter the remaining reads on several quality parameters using VSEARCH, including minimum length, maximum length, maximum expected errors, and maximum number of ambiguous bases. Prior to quality filtering, rename sequence headers to include sample ID and merge all files.

```{code-block} bash
mkdir 3.renamed 4.combined
for fq in 2.demultiplex/*.fastq
do
relabel=${fq/2.demultiplex\//}
vsearch --fastq_filter ${fq} --fastqout ${fq/2.demultiplex/3.renamed/} --relabel ${relabel/.fastq/}.
done
cat 3.renamed/*.fastq > 4.combined/ethanol_comparison_heDNA_combined.fastq
```

After renaming the sequence ID's for each file and all samples merged, we can check the quality and amplicon size range using FastQC. This will be essential to set the minimum and maximum read length thresholds during quality filtering.

```{code-block} bash
mkdir 5.fastqc_pre_filter
fastqc 4.combined/ethanol_comparison_heDNA_combined.fastq -o 5.fastqc_pre_filter
```

![FastQC - Pre-Filter](figures/fastqc_pre_filter.png)

```{code-block} bash
mkdir 6.filtered
vsearch --fastq_filter 4.combined/ethanol_comparison_heDNA_combined.fastq --fastq_maxee 1.0 --fastq_minlen 194 --fastq_maxlen 214 --fastq_maxns 0 --fastqout 6.filtered/ethanol_comparison_heDNA_combined_filtered.fastq --fastaout 6.filtered/ethanol_comparison_heDNA_combined_filtered.fasta
```

```{admonition}
vsearch v2.16.0_macos_aarch64, 16.0GB RAM, 8 cores
https://github.com/torognes/vsearch
Reading input file 100%  
20363664 sequences kept (of which 0 truncated), 3724301 sequences discarded.
```

After quality filtering, we can run FastQC again to check if quality filtering was successful.

```{code-block} bash
mkdir 7.fastqc_post_filter
fastqc 6.filtered/ethanol_comparison_heDNA_combined_filtered.fastq -o 7.fastqc_post_filter
```

![FastQC - Post-Filter](figures/fastqc_post_filter.png)

### 3.5 Dereplication

We will now use the fasta file to dereeplicate the data using VSEARCH, i.e., find unique sequences.

```{code-block} bash
mkdir 8.dereplication
vsearch --derep_fulllength 6.filtered/ethanol_comparison_heDNA_combined_filtered.fasta --relabel uniq. --output 8.dereplication/ethanol_comparison_heDNA_combined_filtered_derep.fasta --sizeout
```

```{admonition}
vsearch v2.16.0_macos_aarch64, 16.0GB RAM, 8 cores
https://github.com/torognes/vsearch
Dereplicating file 6.filtered/ethanol_comparison_heDNA_combined_filtered.fasta 100%  
4114456709 nt in 20363664 seqs, min 194, max 214, avg 202
Sorting 100%
2749957 unique sequences, avg cluster 7.4, median 1, max 1666279
Writing output file 100% 
```

### 3.6 Denoising

Using the unique sequence file, we will denoise the dataset using VSEARCH.

```{code-block} bash
mkdir 9.denoised
vsearch --cluster_unoise 8.dereplication/ethanol_comparison_heDNA_combined_filtered_derep.fasta --sizein --sizeout --relabel denoised. --centroids 9.denoised/ethanol_comparison_heDNA_combined_filtered_derep_denoised.fasta
```

```{admonition}
vsearch v2.16.0_macos_aarch64, 16.0GB RAM, 8 cores
https://github.com/torognes/vsearch
Reading file 8.dereplication/ethanol_comparison_heDNA_combined_filtered_derep.fasta 100%  
23696444 nt in 117334 seqs, min 194, max 214, avg 202
minsize 8: 2632623 sequences discarded.
Masking 100% 
Sorting by abundance 100%
Counting k-mers 100% 
Clustering 100%  
Sorting clusters 100%
Writing clusters 100% 
Clusters: 448 Size min 8, max 3894170, avg 261.9
Singletons: 0, 0.0% of seqs, 0.0% of clusters
```

### 3.7 Chimera removal

Unlike USEARCH, VSEARCH does not automatically remove chimeric sequences during denoising. Hence, we need to execute the following code to accomplish this.

```{code-block} bash
mkdir 10.final
vsearch --uchime3_denovo 9.denoised/ethanol_comparison_heDNA_combined_filtered_derep_denoised.fasta --sizein --nonchimeras 10.final/ethanol_comparison_heDNA_asvs.fasta --relabel asv.
```

```{admonition}
vsearch v2.16.0_macos_aarch64, 16.0GB RAM, 8 cores
https://github.com/torognes/vsearch
Reading file 9.denoised/ethanol_comparison_heDNA_combined_filtered_derep_denoised.fasta 100%  
90837 nt in 448 seqs, min 194, max 214, avg 203
Masking 100% 
Sorting by abundance 100%
Counting k-mers 100% 
Detecting chimeras 100%  
Found 339 (75.7%) chimeras, 109 (24.3%) non-chimeras,
and 0 (0.0%) borderline sequences in 448 unique sequences.
Taking abundance information into account, this corresponds to
44769 (0.3%) chimeras, 16737833 (99.7%) non-chimeras,
and 0 (0.0%) borderline sequences in 16782602 total sequences.
```

We now have created the ASVs for this project.

### 3.8 Count table

Now that we have created the ASVs, we can generate the count table using VSEARCH.

```{code-block} bash
vsearch --usearch_global 6.filtered/ethanol_comparison_heDNA_combined_filtered.fasta --db 10.final/ethanol_comparison_heDNA_asvs.fasta --strand plus --id 0.97 --otutabout 10.final/ethanol_comparison_heDNA_table.txt
```

```{admonition}
vsearch v2.16.0_macos_aarch64, 16.0GB RAM, 8 cores
https://github.com/torognes/vsearch
Reading file 10.final/ethanol_comparison_heDNA_asvs.fasta 100%  
22007 nt in 109 seqs, min 194, max 214, avg 202
Masking 100% 
Counting k-mers 100% 
Creating k-mer index 100% 
Searching 100%  
Matching unique query sequences: 20247244 of 20363664 (99.43%)
Writing OTU table (classic) 100%  
```

## 4. Figure 1: Map of Antarctica

The first figure of the manuscript is a map of Antarctica displaying the locations of the three specimens which were analysed in this experiment. To generate the map, we can run the R script below. To successfully execute the script, several files will need to be downloaded, including:

1. A high resolution Antarctic outline including ice sheets:
   1. source: <https://data.bas.ac.uk/full-record.php?id=GB/NERC/BAS/PDC/01391>
   2. filename: add_coastline_high_res_polygon_v7.2.shp
2. A csv file containing the specimen metadata
   1. filename: specimen_metadata.csv
3. A high resolution bathymetry data file as baselayer:
   1. source: <https://www.gebco.net>
   2. filename: IBCSO_v2_ice-surface.nc
4. A world map with borders for all countries:
   1. source: <https://www.star.nesdis.noaa.gov/data/smcd1/vhp/GIS/TM_World_Borders/>
   2. filename: TM_WORLD_BORDERS-0.3.shp

All files needed to execute the R code can be found in the subdirectory `metadata/figures/figure_1/` within this GitHub repo. Once the figure is generated in R and exported to a pdf file, import figure into Adobe Illustrator to clean up figure for publication.

```{code-block} R
#######################
## MAP OF ANTARCTICA ##
#######################
# set working directory and load libraries
setwd("")
if (!require("pacman")) install.packages("pacman")
pacman::p_load(terra, tidyterra, ggplot2, ggnewscale, patchwork)
require(sf)
library(recolorize)
library(egg)
library(png)
library(ggpubr)
library(viridis)
library(RColorBrewer) 
library(dplyr)
library(hrbrthemes)
library(readxl)
library(ggrepel)
library(pracma)
library(scales)

# read data
ATAshp <- simplifyGeom(vect("add_coastline_high_res_polygon_v7.2/add_coastline_high_res_polygon_v7.2.shp"), tolerance=500)
ATAshp$col <- sapply(ATAshp$surface, function(x){ifelse(x == "land", "grey85", "grey98")})
sponge_dat <- read.csv("specimen_metadata2.csv")
sponge_pts <- project(vect(sponge_dat, geom = c("Longitude1", "Latitude1"), keepgeom = TRUE, crs = "+proj=longlat +ellps=WGS84 +datum=WGS84 +no_defs"), ATAshp)
grat <- project(vect(sf::st_graticule(lon=seq(-175, 180, 5), lat = seq(-85, -60, 5), ndiscr = 5000)), ATAshp)
ibsco <- rast("add_coastline_high_res_polygon_v7.2/IBCSO_v2_ice-surface.nc")
sample_colors <- c("Calcarea" = "lightgoldenrod", "Demospongiae" = "steelblue", "Hexactinellida" = "firebrick")
spp_names <- c(expression(italic("Calcarea")), expression(italic("Demospongiae")), expression(italic("Hexactinellida")))
sponge_pts$Catalog.Number <- as.factor(sponge_pts$Catalog.Number)
xmn <- -1000000
xmx <- 1500000
ymn <- -2500000
ymx <- -1000000
WORLDshp <- simplifyGeom(vect("TM_WORLD_BORDERS-0.3/TM_WORLD_BORDERS-0.3.shp"))
WORLDshpNoATA <- WORLDshp[WORLDshp$ISO3 != "ATA"]

# generate circular masking layer for overview plot
prj <- "+proj=stere +lat_0=-90 +lat_ts=-71 +lon_0=0 +k=1 +x_0=0 +y_0=0 +datum=WGS84 +units=km +no_defs +ellps=WGS84 +towgs84=0,0,0"
little <- st_point(c(0,0)) %>% st_sfc(crs = prj) %>% st_buffer(dist = 4530)
xlim <- sf::st_bbox(little)[c("xmin", "xmax")]*2.5
ylim <- sf::st_bbox(little)[c("ymin", "ymax")]*2.5
encl_rect <- list(cbind(c(xlim[1], xlim[2], xlim[2], xlim[1], xlim[1]), 
                        c(ylim[1], ylim[1], ylim[2], ylim[2], ylim[1]))) %>%
  sf::st_polygon() %>%
  sf::st_sfc(crs = prj)
circleMask <- sf::st_difference(encl_rect, little)

# generate plots
RSR_plot <- ggplot() + 
  geom_spatraster(data = ibsco, show.legend = FALSE) + 
  scale_fill_gradient(low = "#528B8B", high = "#CBDCDC") + 
  geom_spatvector(data = grat, col = "grey50", alpha = 0.2) + 
  geom_spatvector(data = ATAshp, alpha = 1, fill = ATAshp$col, col = 'grey20') + 
  new_scale_fill() + 
  geom_spatvector(data = sponge_pts, aes(shape = Class, fill = Class), alpha = 1, size = 5, stroke = 0.5) + 
  scale_fill_manual(values = sample_colors, labels = c("Calcarea", "Demospongiae", "Hexactinellida") , name = "Sponge Class") +
  guides(fill = guide_legend(override.aes = list(pch=21))) + 
  scale_shape_manual(values = c(21, 22, 23)) + 
  scale_x_continuous(breaks = seq(-180, 180, by = 10)) + 
  coord_sf(crs = crs(ATAshp), expand = FALSE, xlim = c(xmn, xmx), ylim = c(ymn, ymx)) + 
  annotate(geom = "rect", xmin = xmn, xmax = xmx, ymin = ymn, ymax = ymx, fill = NA, col = "grey10") +  
  xlab(NULL) + ylab(NULL) #  

inset_map <- ggplot() + 
  geom_spatvector(data = ATAshp, alpha = 1, fill = ATAshp$col, col = 'grey20') + 
  geom_rect(aes(xmin = xmn, xmax = xmx, ymin = ymn, ymax = ymx), fill = "dark red", col = "black", alpha = 0.2) + 
  theme(panel.grid.minor = element_blank(), panel.grid.major = element_blank(), panel.background = element_blank(), plot.background = element_blank(), panel.border = element_blank(), axis.line = element_blank(), axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(), axis.title.x = element_blank(), axis.title.y = element_blank(), plot.title = element_blank()) 

#output to pdf
pdf("RSR_spongemap_v2.pdf", width = 8, height = 5.5) 
RSR_plot + annotation_custom(ggplotGrob(inset_map), xmin = xmx - (xmx-xmn)/3, xmax = xmx, ymin = ymx - (ymx-ymn)/3, ymax = ymx + 280000) 
dev.off() 
```

## 5. Figure 2: DNA concentration and purity comparison
